### PR TITLE
Implement Back4App Integration and Topic-Based User Matching for Chat

### PR DESCRIPTION
**Pull request by aywebe**
Implemented Back4App backend integration to facilitate user login processes. Enhanced the platform by introducing a feature that allows users to select conversation topics for matching purposes; these preferences are now maintained in the backend, setting the stage for upcoming chat functionalities. Ongoing efforts are focused on developing the chat system, which aims to pair users based on shared topic interests and generate a unique chat ID for each successful match.

### CODE REVIEW

File: src/components/atoms/LogoutFunction.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        return true;
+      } else {
+        alert(`Error! ${error.message}`);
+        return false;
+      }
+    } catch (error) {
+      alert(`Error! ${error.message}`);
+      return false;
+    }
+  }
```
Comment (igawaclawska) [inline src/components/atoms/LogoutFunction.jsx:24]: Is the Button responsible for the Log-Out process? 
What do you think of this approach, @mircealungu ?

File: src/components/molecules/FieldMolecule.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// FieldMolecule.jsx component - a wrapper for InputField with label and error message
+
+import InputField from "../atoms/InputField";
+
+const FieldMolecule = ({
```
Comment (igawaclawska) [inline src/components/molecules/FieldMolecule.jsx:5]: Rename with `Field`

File: src/components/pages/NewChat.jsx [hunk 3]
```diff
@@ -1,54 +1,83 @@
...
+
+  // Function that will return current user and also update current username
+  const getCurrentUser = async function () {
+    const currentUser = await Parse.User.current();
+    // Update state variable holding current user
+    setCurrentUser(currentUser);
+    return currentUser;
+  };
+
+  // function to retreive topics and create list of dictionaries (incl. topic_id and topic_name)
```
Comment (igawaclawska) [inline src/components/pages/NewChat.jsx:38]: Nice that you have added comments to your code! :)

**Review by igawaclawska — CHANGES_REQUESTED** [review]
Good job overall!
I have added a few comments. Take a look at them and apply changes you agree with. Let us know in the comments once you are done.

iga and @mircealungu

File: src/components/molecules/FieldMolecule.jsx (hunk 2 above)
Comment (aywebe) [reply src/components/molecules/FieldMolecule.jsx:5]: Done :)

File: src/components/organisms/AuthFormOrganism.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
+// AuthFormOrganism.jsx organism component  
+
+import FieldMolecule from "../molecules/FieldMolecule";
+import GenericButton from "../atoms/GenericButton";
+
+const AuthFormOrganism = ({
+  onSubmit, // Function to execute on form submission
```
[the remark this answers is not in the document]
Comment (aywebe) [reply src/components/organisms/AuthFormOrganism.jsx:7]: Done :)

**Review by igawaclawska — APPROVED** [review]

**Pull request merged**